scene/main/viewport.cpp

```diff
@@ -2949,6 +2930,156 @@ bool Viewport::_sub_windows_forward_input(const Ref<InputEvent> &p_event) {
 	return true;
 }
 
+void Viewport::_update_mouse_over() {
+	// Update gui.mouse_over and gui.subwindow_over in all Viewports.
+	// Send necessary mouse_enter/mouse_exit signals and the NOTIFICATION_VP_MOUSE_ENTER/NOTIFICATION_VP_MOUSE_EXIT notifications for every Viewport in the SceneTree.
+
+	if (is_attached_in_viewport()) {
+		// Execute this function only, when it is processed by a native Window or a SubViewport, that has no SubViewportContainer as parent.
+		return;
+	}
+
+	if (get_tree()->get_root()->is_embedding_subwindows() || Object::cast_to<SubViewport>(this)) {
```

Comment:
This reminds me of something we discussed in another PR, that it felt awkard to check the type of `this` that way. However, I don't remember where that was nor the decision. Do you know what I'm talking about?

Reply:
Yes I remember, that was https://github.com/godotengine/godot/pull/77923#discussion_r1222519549
I will look into making this less awkward.

Reply:
I have implemented this in a different way.